File: Assignment_2/Assignment2_4.py
"""-------------------------------------------------------------------------------------------
                          Assignment2_4
                    (Student name - Vaishali Jorwekar)
                     Python by Marvellous Infosystems
---------------------------------------------------------------------------------------------
Problem statement : This program accepts a number from user and 
                    displays the addition of it's factors
------------------------------------------------------------------------------------------"""
from Arithmatic import Add
#----------------------------------------------------------------------------------------
# This function calculates addition of factors of given number and returns 
# all factors of given number and addition of factors
#---------------------------------------------------------------------------------------
def calculate_addition_factors(number):
    result = 0
    factors=[]
    for num in range(1,number):
        if (number % num == 0) :
            result = Add(result,num)
            factors.append(num)
    return factors,result
#------------------------------------------------------------------------------------------
# This function accepts a input number from user and calls factorial function
#------------------------------------------------------------------------------------------
def main():
    number=int(input("Enter number :"))
    factors,result=calculate_addition_factors(number)
    print("--------------------------------------------------------------")
    print(f"Addition of all factors {factors} of {number} is :",result)
    print("--------------------------------------------------------------")

#-----------------------------------------------------------------------------------
# This is main entry point of the program
#-----------------------------------------------------------------------------------
if __name__=="__main__":
    main()
#-----------------------------------------------------------------------------------

File: JavaPractice/Assignment_2/Assignment2_4.java

/*-------------------------------------------------------------------------------------------
                           Assignment2_4
                    (Student name - Vaishali Jorwekar)
                     Python by Marvellous Infosystems
---------------------------------------------------------------------------------------------
Problem statement : This program accepts a number from user and 
                    displays the addition of it's factors
------------------------------------------------------------------------------------------*/
import java.util.Scanner;




public class Assignment2_4 {
	
	public static void findFactors(int number) {
		System.out.println("Factors of "+number+" are:");
		for(int cnt=1;cnt<=number;cnt++) {
			if(number%cnt==0) {
				System.out.print(cnt+" ");
			}
		}
	}

	public static void main(String[] args) {
		Scanner inputObj=new Scanner(System.in);
		System.out.println("Enter the number:");
		int number=inputObj.nextInt();
		findFactors(number);
		inputObj.close();
		

	}

}
